opendevin/sandbox/docker/ssh_box.py

```diff
@@ -41,6 +42,14 @@
 elif hasattr(os, 'getuid'):
     USER_ID = os.getuid()
 
+def split_bash_commands(bash_string):
+    # This regex matches an escaped newline (i.e., a backslash followed by a newline)
```

Comment:
Where do we expect this escaped newline to come from? Do we generally escape any newline that comes from the agent? How do we distinguish if a newline is between commands, or within command?

For example: `echo "hello\n world"` vs. `ls\npwd`

Reply:
Good catch--the former does break 😢 

Will work on this

Reply:
Fixed, and added a test case!

Does codeact actually output multi-line commands like ls\npwd? They seem very hard to support

Reply:
Yeah it's fair. Hopefully my command splitter (courtesy of GPT-4) works well enough!